### PR TITLE
refactored the way the UI is presented

### PR DESCRIPTION
**Pull request by Malvolio**
changed a lot of state to context
factored the login management to a parent component
factored the "top button" out
added a new login state, "undefined", so the user does not see the login button after she's logged in.

### CODE REVIEW

File: frontend/src/StealthHome.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from "react";
```
Comment (tstanis-snickerbockens) [inline frontend/src/StealthHome.tsx:1]: rename StoryHome

File: frontend/src/StealthHome.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  children,
+  onClick,
+}) => {
+  const classes = styles();
+
+  return (
+    <Button
+      variant="contained"
+      color="inherit"
+      id="loginState"
```
Comment (tstanis-snickerbockens) [inline frontend/src/StealthHome.tsx:66]: is this id necessary?

File: frontend/src/StealthHome.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        const provider = new firebase.auth.GoogleAuthProvider();
+        firebase.auth().signInWithRedirect(provider);
+      } else {
+        firebase.auth().signOut();
+        setUser(null);
+      }
+    };
+
+    setTopButton(
+      userTopButton ||
```
Comment (tstanis-snickerbockens) [inline frontend/src/StealthHome.tsx:115]: this nested logic is hard to follow, consider breaking into several statements.

File: frontend/src/WelcomePage.tsx [hunk 4]
```diff
@@ -13,14 +13,16 @@ const useStyles = makeStyles(() => ({
...
+export const WelcomePage: React.FC<{ waiting?: boolean }> = ({ waiting }) => {
   const classes = useStyles();
 
   return (
     <div className={classes.splashScreen}>
       <Typography variant="h4">Welcome to</Typography>
-      <Typography variant="h2">Stealth Health : Telemedicine for the Elderly</Typography>
-      <Typography>Please log in.</Typography>
+      <Typography variant="h2">
+        Stealth Health : Telemedicine for the Elderly
```
Comment (tstanis-snickerbockens) [inline frontend/src/WelcomePage.tsx:23]: Story

File: frontend/src/StealthHome.tsx (hunk 1 above)
Comment (Malvolio) [reply frontend/src/StealthHome.tsx]: Done.

File: frontend/src/StealthHome.tsx (hunk 3 above)
Comment (Malvolio) [reply frontend/src/StealthHome.tsx]: Removed.

File: frontend/src/StealthHome.tsx (hunk 2 above)
Comment (Malvolio) [reply frontend/src/StealthHome.tsx]: Done.

File: frontend/src/WelcomePage.tsx (hunk 4 above)
Comment (Malvolio) [reply frontend/src/WelcomePage.tsx]: Done.

File: frontend/src/ClinicianApp.tsx [hunk 5]
```diff
@@ -98,93 +94,31 @@ const MainClinicianPanel = ({ onVisit, user }: MainClinicianPanelProps) => {
...
-        </Toolbar>
-      </AppBar>
-      {pageComponent}
-      <AppBar
-        position="fixed"
-        color="primary"
-        className={classes.bottomBar}
-      ></AppBar>
-    </div>
+  return encounterId ? (
```
Comment (tstanis-snickerbockens) [inline frontend/src/ClinicianApp.tsx:114]: not sure why the ternary for a long statement like this.
